pmd-java/src/main/resources/category/java/performance.xml

```diff
@@ -26,6 +26,8 @@ It is much better to use one of the type-specific toString() methods instead.
                 <value>
 <![CDATA[
 //AdditiveExpression/PrimaryExpression/PrimaryPrefix/Literal[@Image='""' and not(ancestor::Annotation)]
+|
+//AdditiveExpression/PrimaryExpression/PrimaryPrefix/Name[contains(string-join(//VariableInitializer/Expression/PrimaryExpression/PrimaryPrefix/Literal[@Image='""']/../../../../../VariableDeclaratorId/@Name,','),@Image)]
```

Comment:
* Using `//VariableInitializer` searches in the whole file - that might find variables with the same name declared somewhere else
* Using `contains` to compare is not exact - e.g. it will assume that "modelDebug" is the same variable as "b", because it contains "b". We need to compare it with `=`, e.g.
`... PrimaryPrefix/Name[@Image = .../VariableDeclaratorId/@Name]` (warn: this is pseudo code which won't work, but I hope you get the idea)